### DISCUSSION TITLE
links in `quarto preview` hangs when file created by `pre-render` script already exists before preview

### DISCUSSION THREAD

**Discussion by liyongg** [quarto-dev/quarto-cli#6075] [Q&A]
#### Context
For my project I use a Quarto Book and its pre-render options to manipulate serval files, including tex.files. For example, for PDF output I would like variables in the `_variables.yml` to display in the header, such that a change in `_variables.yml` is enough to change the header in the PDF. All of the pre-render works well for the PDF and the output is as I had in mind.

However, at the same time I want to preview the contents (not neccesarily header) html output using `quarto preview`. I noticed that clicking on chapters different from the `index.qmd` resulted in the preview hanging and the reloading the homepage/index. I believe I have found a reprex, but execuse me if it still is not 100% reproducible. Also, I don't know if this is by design so I did not want to create an issue unnecessarily, but will happily make an issue if this indeed is one. While making the reprex, I might have found a solution. (Found this after spending half an hour on typing this)

#### Errors/warnings/logs
When clicking on another chapter to view it, in the IDE terminal I see the lines
```bash
GET: /intro.html
WARNING: connection closed before message completed
```
In the browser localhost, the render window pops up and says:
```bash
projectContext: Found Quarto project in C:\Tools\testbooklet
```
After this text shows up, the terminal appends with:
```bash
GET: /
```
As it then reloads/refreshed to the index.qmd/homepage.

#### Steps to reproduce
1. Create a Quarto book project (testbooklet).
2. Create a pre-render.R file as such in the root.
    ```r
    conn <- file("design.tex")
    writeLines('placeholder', con = conn)
    close(conn)
    ```
3. Modify `_quarto.yml` by adding these options to the existing options
    ```yml
    project:
      pre-render: pre-render.R
    
    format:
      pdf:
        include-in-header:
          - design.tex
    ```
4. Run `quarto preview` once. Everything should work as normal and chapters should be clickable and loads when clicked. Render window simply shows: `Windows: using code page 1252`
5. Close the preview. Run `quarto preview` once more. Now, `design.tex` already exists before pre-render scripts are executed.
6. Preview only allows preface to load.
7. Any subsequent attempts at `quarto preview` result in similar behaviour. Deleting `design.tex` at root helps.

#### Solution
I tried a post-render script in which the created file is automatically deleted:
```r
file.remove('design.tex')
```
This seems to work! But I am not sure this is the go-to solution.

#### Question
Would it be possible for this behaviour to be somehow better captured in the logging? And specifically I mean the render window that pops up. The current logging of 'projectContext: Found Quarto project in C:\Tools\testbooklet
' did not bring me really far in finding a solution that at least works for me. Then again, I am not sure how you would want to design this if the link/chapter you click on does not load. Break the entire preview? ...

#### Software
RStudio: RStudio 2023.06.0+421
OS: Windows10 64bit
Browser: Edge113.0.1774.35 (Official) (64-bits) (tested it in other browsers, same behaviour)

**Comment by mcanouil** [comment]
I believe, the right answer is to update your pre-render script to do what you want or as you did using a post-render script.

Quarto can't know what you are creating or not, thus what need to be deleted or not.

```r
if (file.exists("design.tex")) file.remove("design.tex")

conn <- file("design.tex")
writeLines('placeholder', con = conn)
close(conn)
```
